web/templates/message_controls.hbs

```diff
@@ -36,11 +36,11 @@
     {{/unless}}
 
     <div class="message_failed {{#unless msg.failed_request}}hide{{/unless}}">
-        <div class="message_control_button failed_message_action" data-tippy-content="{{t 'Retry' }}">
+        <div class="message_control_button failed_message_action tippy-zulip-tooltip" data-tippy-content="{{t 'Retry' }}">
             <i class="fa fa-refresh refresh-failed-message" aria-label="{{t 'Retry' }}" role="button" tabindex="0"></i>
         </div>
 
-        <div class="message_control_button failed_message_action" data-tippy-content="{{t 'Cancel' }}">
+        <div class="message_control_button failed_message_action tippy-zulip-tooltip" data-tippy-content="{{t 'Cancel' }}">
```

Comment:
We appear to have custom `tippyjs.js` code for these elements; I think it might be more appropriate to either migrate them to use `data-tippy-template-id` or to adjust the code below to support getting the content from a `data-tippy-content` if available.  (I don't think we intend to have multiple tooltip delegations on the same element).

```
    delegate("body", {                                                         
        target: ".message_control_button",                                     
        // This ensures that the tooltip doesn't                               
        // hide by the selected message blue border.                           
        appendTo: () => document.body,                                         
        // Add some additional delay when they open                            
        // so that regular users don't have to see                             
        // them unless they want to.                                           
        delay: LONG_HOVER_DELAY,                                               
        onShow(instance) {                                                     
            // Handle dynamic "starred messages" and "edit" widgets.           
            const $elem = $(instance.reference);                               
            const $template = $("#" + $elem.attr("data-tooltip-template-id")); 
            instance.setContent(parse_html($template.html()));                 
        },                                                                     
    });                                                                        
```

Reply:
I have made adjustments to the code snippet you provided above to accommodate for getting content from `data-tippy-content` if available.